controllers/admin/AdminAjaxPsAccountsController.php

```diff
@@ -204,4 +205,37 @@ public function ajaxProcessGetContext()
             );
         }
     }
+
+    /**
+     * @return void
+     *
+     * @throws Exception
+     */
+    public function ajaxProcessMigrateOrCreateIdentityV8()
+    {
+        header('Content-Type: text/json');
+        $shopId = Tools::getValue('shop_id', null);
+
+        try {
+            if (!$shopId) {
+                throw new Error('Shop ID is required for migration or creation.');
```

Comment:
Any reason to throw an `Error` here ?
Moreover it won't be catched by your catch later on more recent versions of PHP.